pkg/helmshim/helm_test.go

```diff
@@ -16,12 +17,38 @@ func TestRenderChart(t *testing.T) {
 	input := bytes.NewBufferString(`{ "items": [{ "apiVersion": "v1", "kind": "ConfigMap", "metadata": { "name": "test-cm", "annotations": { "eno.azure.io/input-key": "foo" } } }] }`)
 	i, err := function.NewInputReader(input)
 	require.NoError(t, err)
-
 	err = RenderChart(WithInputReader(i), WithOutputWriter(o), WithChartPath("fixtures/basic-chart"))
 	require.NoError(t, err)
+	err = o.Write()
+	require.NoError(t, err)
 	assert.Equal(t, "{\"apiVersion\":\"config.kubernetes.io/v1\",\"kind\":\"ResourceList\",\"items\":[{\"apiVersion\":\"v1\",\"data\":{\"input\":\"{\\\"apiVersion\\\":\\\"v1\\\",\\\"kind\\\":\\\"ConfigMap\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"eno.azure.io/input-key\\\":\\\"foo\\\"},\\\"name\\\":\\\"test-cm\\\"}}\",\"inputResourceName\":\"test-cm\",\"some\":\"value\"},\"kind\":\"ConfigMap\",\"metadata\":{\"name\":null}},{\"apiVersion\":\"somegroup.io/v9001\",\"kind\":\"ATypeNotKnownByTheScheme\",\"metadata\":{\"name\":\"foo\"}}]}\n", output.String())
 }
 
+func TestRenderChartWithDefaultOutputWriter(t *testing.T) {
+	// Save the original stdout and redirect it to a pipe.
+	oldStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
```

Comment:
This is clever, but I'm wondering if it might have some strange interactions with the test runner, since it might write to stdout. I wouldn't block on this, but would slightly prefer to define a global var in this package like `var stdout = os.Stdout`, use it as the output writer default, and override only that value from the tests.

Reply:
agree with the concern, i change to use package level stdout to create the output writer to simulate the behavior of default output writer which uses os.Stdout.

Reply:
discuss offline, remove the test